File: hrms/src/main/java/com/obss/hrms/response/GetAdvertisementResponse.java
package com.obss.hrms.response;

import com.obss.hrms.entity.AdvertisementStatue;
import com.obss.hrms.entity.HumanResourceEntity;
import com.obss.hrms.entity.PersonalSkill;

import java.time.LocalDate;
import java.util.List;

public record GetAdvertisementResponse(
        String id,
        String advCode,
        String title,
        String jobDescription,
        LocalDate activationTime,
        LocalDate offDate,
        AdvertisementStatue advertisementStatue,
        HumanResourceEntity humanResourceEntity,
        List<PersonalSkill> personalSkills
) {
}

File: hrmsFe/src/app/model/getAdvertisementResponse.ts
import {GetHumanResourceEntityResponse} from "./getHumanResourceEntityResponse";
import {GetPersonalSkillResponse} from "./getPersonalSkillResponse";

export interface GetAdvertisementResponse{

  id:string,

  advCode:string,

  title:string,

  jobDescription:string,

  activationTime:Date,

  offDate:Date,

  advertisementStatue:String, // TODO buna da bak

  humanResourceEntity:GetHumanResourceEntityResponse,

  personalSkills: GetPersonalSkillResponse[]
}
